src/NuGet.Clients/NuGet.CommandLine/GetProjectsReferencingProjectJsonFiles.targets

```diff
@@ -16,8 +16,14 @@
                 Condition=" '$(MSBuildProjectFullPath)' != '$(EntryPointProject)' " />
         </ItemGroup>
 
+        <!-- Filter out project references that specify ReferenceOutputAssembly=false -->
+        <ItemGroup>
+          <ValidProjectInputForNuGet Include="@(ProjectReference)" 
+              Condition=" %(ProjectReference.ReferenceOutputAssembly) != 'false' AND %(ProjectReference.ReferenceOutputAssembly) != 'off' AND %(ProjectReference.ReferenceOutputAssembly) != '0'" />
```

Comment:
Why the extra conditions? My understanding is MSBuild does some magic so Boolean comparisons might do what you want. In any case, the equivalent condition in the core targets is just:

          <Output TaskParameter="TargetOutputs" ItemName="_ResolvedProjectReferencePaths" Condition="'%(_MSBuildProjectReferenceExistent.ReferenceOutputAssembly)'=='true'"/>